Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py

```diff
@@ -0,0 +1,264 @@
+            ignore_filters.append({'Value': event_id, 'Comparison': 'NOT_EQUALS'})
+
+        filters['Id'] = ignore_filters
+
+    if filters:
+        # We send kwargs because passing Filters=None to get_findings() tries to use a None value for filters,
+        # which raises an error.
+        kwargs['Filters'] = filters
+
+    count = 0
+
+    while True:
+        # The API only allows a maximum of 100 results per request. Using more raises an error.
+        if limit and limit - count < 100:
+            kwargs['MaxResults'] = limit - count
+
+        else:
+            kwargs['MaxResults'] = 100
+
+        response = client.get_findings(**kwargs)
+        result = response.get('Findings', [])
+        count += len(result)
+        yield result
+
+        if 'NextToken' in response and (limit == 0 or count < limit):
+            kwargs['NextToken'] = response['NextToken']
+
+        else:
+            break
+
+
+def fetch_events(client: boto3.client, last_run: dict,
+                 first_fetch_time: dt.datetime | None, limit: int = 0) -> tuple[list[dict], Exception | None]:
+    """
+    Fetch events from AWS Security Hub and send them to XSIAM.
+
+    Args:
+        client (boto3.client): Boto3 client to use.
+        last_run (dict): Dict containing the last fetched event creation time.
+        first_fetch_time (datetime | None, optional): In case of first fetch, fetch events from this datetime.
+        limit (int): Maximum number of events to fetch. Defaults to 0 (no limit).
+    """
+    if last_run.get('last_update_date'):
+        start_time = parse_date_string(last_run['last_update_date'])
+
+    else:
+        start_time = first_fetch_time
+
+    id_ignore_list: list = last_run.get('last_update_date_finding_ids', [])
+
+    events = []
+    error = None
+
+    try:
+        for events_batch in get_events(client=client, start_time=start_time,
+                                       id_ignore_list=id_ignore_list, limit=limit):
+            events.extend(events_batch)
+
+    except Exception as e:
+        demisto.error(f'Error while fetching events: {e}')
+        error = e
+
+    # --- Set next_run data ---
+    # Since the time filters seem to be equal or greater than (which results in duplicate from the last run),
+    # we add findings that are equal to 'last_finding_update_time' and filter them out in the next fetch.
+    if events:
+        last_update_date = events[-1].get(TIME_FIELD)
+        demisto.info(f'Fetched {len(events)} findings.\nUpdate time of last finding: {last_update_date}.')
+
+        ignore_list: list[str] = []
+        last_update_date = events[-1].get(TIME_FIELD)
+
+        for event in events:
+            event['_time'] = event[TIME_FIELD]
+
+            if event[TIME_FIELD] == last_update_date:
+                ignore_list.append(event['Id'])
+
+        next_run = {
+            'last_update_date': last_update_date,
+            'last_update_date_finding_ids': ignore_list,
+        }
+
```

Comment:
let's move all this to another function to make it more readable
and here it would look like:
```
    if events:
        next_run = get_next_run(events)